contracts/staking/DefaultTargetDispenserL2.sol

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "../interfaces/IToken.sol";
+
+interface IServiceStakingFactory {
+    function verifyInstance(address instance) external view returns (bool);
+}
+
+interface IServiceStaking {
+    function deposit(uint256 amount) external;
+}
+
+error TargetRelayerOnly(address messageSender, address l2MessageRelayer);
+error WrongSourceChainId(uint256 sourceChainId, uint256 l1SourceChainId);
+error SourceProcessorOnly(address sourceProcessor, address l1SourceProcessor);
+error ReentrancyGuard();
+error OwnerOnly(address sender, address owner);
+error ZeroAddress();
+error ZeroValue();
+
+abstract contract DefaultTargetDispenserL2 {
+    event ServiceStakingTargetDeposited(address indexed target, uint256 amount);
+    event ServiceStakingAmountWithheld(address indexed target, uint256 amount);
+    event ServiceStakingRequestQueued(bytes32 indexed queueHash, address indexed target, uint256 amount, uint256 batchNonce);
+    event ServiceStakingParametersUpdated(uint256 rewardsPerSecondLimit);
+    event MessageSent(uint256 indexed sequence, address indexed messageSender, address indexed l1Processor, uint256 amount);
+    event MessageReceived(address indexed messageSender, uint256 chainId, bytes data);
+    event WithheldAmountSynced(address indexed sender, uint256 amount);
+    event Paused();
+    event Unpaused();
+
+    // Gas limit for sending a message to L1
+    uint256 public constant GAS_LIMIT = 100_000;
+    // OLAS address
+    address public immutable olas;
+    // Proxy factory address
+    address public immutable proxyFactory;
+    // Owner address (Timelock or bridge mediator)
+    address public immutable owner;
+    // L2 Relayer address that receives the message across the bridge from the source L1 network
+    address public immutable l2MessageRelayer;
+    // Source processor address on L1 that is authorized to propagate the transaction execution across the bridge
+    address public immutable l1SourceProcessor;
+    // Source processor chain Id
+    uint256 public immutable l1SourceChainId;
+    // Amount of OLAS withheld due to service staking target invalidity
+    uint256 public withheldAmount;
+    // Nonce for each staking batch
+    uint256 public stakingBatchNonce;
+    // rewardsPerSecondLimit
```

Comment:
```suggestion

```

Reply:
rewardsPerSecondLimit was removed after the introduction of the verifier logic in the factory